src/mscorlib/corefx/System/Globalization/CompareInfo.Unix.cs

```diff
@@ -312,5 +312,12 @@ private static byte[] GetNullTerminatedUtf8String(string s)
 
             return buffer;
         }
+        
+        private bool CanUseOrdinal(string source, string target, CompareOptions options)
+        {
+            return m_isAsciiEqualityOrdinal && CanUseAsciiOrdinalForOptions(options) && 
+                   source.IsAscii() && !source.IsFastSort() && 
+                   target.IsAscii() && !target.IsFastSort();
```

Comment:
@stephentoub  you can look at following sources to know what IsFastSort is doing:

https://github.com/dotnet/coreclr/blob/master/src/classlibnative/bcltype/stringnative.cpp#L211
https://github.com/dotnet/coreclr/blob/master/src/vm/object.cpp#L2370

basically the overhead of IsFastSort is minimal as it'll not walk the string again and will just use the flag stored in the string object (exactly as IsAscii).